### PR TITLE
[plugin.video.sito.tv] v1.3.1

### PR DESCRIPTION
**Pull request by sito-tv-org**
### Description

SiTo.tv is a plugin for Kodi for streaming public domain & other non-copyrighted moves and TV Shows using BitX Player and BitTorrent peer 2 peer communication protocol.

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by sito-tv-org** [comment]
Hi, any update on this ?

File: plugin.video.sito.tv/resources/lib/utils.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            return "linux"
+        elif xbmc.getCondVisibility("system.platform.xbox"):
+            return "xbox"
+        elif xbmc.getCondVisibility("system.platform.windows"):
+            return "windows"
+        elif xbmc.getCondVisibility("system.platform.osx"):
+            return "darwin"
+        elif xbmc.getCondVisibility("system.platform.ios"):
+            return "ios"
+        elif xbmc.getCondVisibility("system.platform.atv2"):
```
Comment (enen92) [inline plugin.video.sito.tv/resources/lib/utils.py]: no longer exists

File: plugin.video.sito.tv/resources/lib/views.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+            kodi_go_back()
+            return
+        except Exception as ex:
+            log("BitX magnet start error: %s" % ex, level=xbmc.LOGERROR)
+
+    notice(_lang(33023), _lang(33022), 2500)
+    kodi_go_back()
+
+
+def kodi_go_back():
```
Comment (enen92) [inline plugin.video.sito.tv/resources/lib/views.py]: why do you need this? Likely you are wrongly creating listitems, `isFolder=False` will not create browsable vfs directories

File: plugin.video.sito.tv/resources/lib/views.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+            'info': {'plot': _lang(33005)}
+        }, 'tvshows'),
+        prepare_list_item({
+            'label': '[COLOR {}]{}[/COLOR]'.format(COLOR_WHITE, _lang(33006)),
+            'path': plugin.url_for('action_custom_action', action='Addon.OpenSettings(%s)' % PLUGIN_ID),
+            'info': {'plot': _lang(33007)}
+        }, 'settings')
+    ]
+    log("index route")
+    return plugin.finish(result, view_mode=ViewMode.IconWall)
```
Comment (enen92) [inline plugin.video.sito.tv/resources/lib/views.py]: no idea what this does but you are not allowed to change skin views

File: plugin.video.sito.tv/resources/settings.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <setting label="Email" visible="false" type="text" id="user_email" default="" />
+        <setting label="Send authorization email" visible="false" type="action" action="RunScript(kodi.sito.tv, custom_action, authorization)" subsetting="true"/>
+        <setting label="Auth code from email" visible="false" type="text" id="auth_code" option="hidden" enable="!eq(-2,)" default="" subsetting="true"/>
+        <setting label="Auth Result Token" type="text" id="auth_token" default="" visible="false" subsetting="true"/>
+
+        <setting label="Last update check time" type="number" id="last_update_check_time" default="0" visible="false" subsetting="true"/>
+    </category>
+
+    <category label="Plugin customization">
+        <setting id="api_url" type="text" label="Custom data API URL" default="http://vm160.siriomedia.com"/>
```
Comment (enen92) [inline plugin.video.sito.tv/resources/settings.xml]: custom api url as a user settings looks sketchy as hell...

**Comment by sito-tv-org** [comment]
@enen92 Thank you for the feedback. We'll update the code changes requested.

To address some of the concerns : 

> I'm personally not confortable into accepting a browsing torrent streaming application to our repository. 

This is one of the issues we pondered when considering the the PR. Somewhere in the guidelines said the addon must not violate any copyright laws, and if we are in doubt we could check with you. Since public domain media is non-copyrighted we assumed it wouldn't be an issue.

> The API url is just a generic VM in which there is no control, at anypoint in time the content might change with ilegal videos.

The idea is to have a single REST API as an interface to the public domain media. 
The API is developed by us, and we have no intention of hosting illegal torrents.

**Comment by enen92** [comment]
After internal discussions we decided not to accept nor distribute this plugin in our repository. Thanks for the contribution though

**Pull request closed without merging**